runtime/tests/automl_runtime/sklearn/one_hot_transformer_test.py

```diff
@@ -0,0 +1,126 @@
+        self.expected_categorical_X_indicator = [[1, 0, 0, 1], [0, 1, 0, 3], [0, 1, 0, 2]]
+
+        self.categorical_X_with_none = [['Male', 1], [np.nan, 2], [None, 3], [pd.NA, 1]]
+        # np.nan == None, but pd.na is a different item (unknown?) so we fill a different column for both
+        self.expected_categorical_X_with_none = [[1, 0, 0, 1],[0, 1, 0, 2],[0, 1, 0, 3], [0, 0, 1, 1]]
+
+    def test_fit_and_transform(self):
+        numerical_output = OneHotTransformer().fit_transform(self.numeric_X)
+        self.assertTrue(isinstance(numerical_output, sparse.csr_matrix))
+        np.testing.assert_almost_equal(numerical_output.toarray(), self.expected_numeric_x)
+
+        categorical_output = OneHotTransformer().fit_transform(self.categorical_X)
+        self.assertTrue(isinstance(categorical_output, sparse.csr_matrix))
+        np.testing.assert_almost_equal(categorical_output.toarray(), self.expected_categorical_X_value) # default is value
+        categorical_output = OneHotTransformer(handle_unknown="indicator").fit_transform(self.categorical_X)
+        np.testing.assert_almost_equal(categorical_output.toarray(), self.expected_categorical_X_indicator)
+ 
+        categorical_output_with_none = OneHotTransformer(handle_unknown="indicator").fit_transform(self.categorical_X_with_none)
+        self.assertTrue(isinstance(categorical_output_with_none, sparse.csr_matrix))
+        np.testing.assert_almost_equal(categorical_output_with_none.toarray(), self.expected_categorical_X_with_none)
+
+    def test_transform_after_fit(self):
+        transformer = OneHotTransformer()
+        transformer.fit(self.numeric_X)
+        numerical_output = transformer.transform(self.numeric_X)
+        self.assertTrue(isinstance(numerical_output, sparse.csr_matrix))
+        np.testing.assert_almost_equal(numerical_output.toarray(), self.expected_numeric_x)
+
+        transformer = OneHotTransformer()
+        transformer.fit(self.categorical_X)
+        categorical_output = transformer.transform(self.categorical_X)
+        self.assertTrue(isinstance(categorical_output, sparse.csr_matrix))
+        np.testing.assert_almost_equal(categorical_output.toarray(), self.expected_categorical_X_value) # default is value
+
+        transformer = OneHotTransformer(handle_unknown="indicator")
+        transformer.fit(self.categorical_X)
+        categorical_output = transformer.transform(self.categorical_X)
+        np.testing.assert_almost_equal(categorical_output.toarray(), self.expected_categorical_X_indicator)
+
+
+        transformer = OneHotTransformer(handle_unknown="indicator")
+        transformer.fit(self.categorical_X_with_none)
+        categorical_output_with_none = transformer.transform(self.categorical_X_with_none)
+        self.assertTrue(isinstance(categorical_output_with_none, sparse.csr_matrix))
+        np.testing.assert_almost_equal(categorical_output_with_none.toarray(), self.expected_categorical_X_with_none)
```

Comment:
I'm wondering about what value this test adds given that it is very similar to the above one. Wdyt?

Reply:
I guess since I have separate functions for fit and transform (other than fit_transform), I added these tests for code coverage. I reasonably could just call the fit and transform functions inside fit_transform and delete these tests. I think that would actually make more sense anyway.